.github/workflows/test-runner-pr-integration.yml

```diff
@@ -0,0 +1,103 @@
+    - name: Cache Gradle packages
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
+        restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+
+    - name: Get Vault token
+      id: vault-token-step
+      env:
+        VAULT_ADDR: https://clotho.broadinstitute.org:8200
+      run: |
+        VAULT_TOKEN=$(docker run --rm --cap-add IPC_LOCK \
+          -e "VAULT_ADDR=${VAULT_ADDR}" \
+          vault:1.1.0 \
+          vault write -field token \
+            auth/approle/login role_id=${{ secrets.VAULT_APPROLE_ROLE_ID }} \
+            secret_id=${{ secrets.VAULT_APPROLE_SECRET_ID }})
+        echo ::set-output name=vault-token::$VAULT_TOKEN
+        echo ::add-mask::$VAULT_TOKEN
+    - name: Set permissions and render test configurations
+      run: |
+        chmod +x render_config.sh
+        ./render_config.sh dev ${{ steps.vault-token-step.outputs.vault-token }}
+        chmod +r rendered
+
+    # Spin up a local server.
+    - name: Run local server (in background)
+      run: ./gradlew bootRun &
+    - name: Wait for local server to start
+      # With a timeout of 60 seconds, try to connect to localhost:8080 every 1 second.
+      run: timeout 60 bash -c 'until echo > /dev/tcp/localhost/8080; do sleep 1; done'
+
+    # We can also set the env var in the GH action script to specify which server we are testing against.
+    # This will be particularly helpful once we switch over to testing against any test env -- then it's just a change to the env var.
+    # Please refer to https://github.com/DataBiosphere/terra-test-runner#override-the-server-from-the-command-line for more information.
+    - name: Run the Basic Integration test suite
+      run: |
+        cd workspace-manager-clienttests
+        echo "Test Runner config"
+        ./render-config.sh ${{ steps.vault-token-step.outputs.vault-token }}
+        echo "Running test suite"
+        export TEST_RUNNER_SERVER_SPECIFICATION_FILE="workspace-local.json"
+        ./gradlew runTest --args="suites/BasicIntegration.json /tmp/TestRunnerResults"
+
+    - name: Upload test results to Google Bucket
+      if: ${{ always() }}
+      run: |
+        cd workspace-manager-clienttests
+        echo "Upload test results to Google Bucket"
+        ./gradlew uploadResults --args="CompressDirectoryToBucket.json /tmp/TestRunnerResults"
+
+    - name: Upload test results to BigQuery Test Runner Dataset
+      if: ${{ always() }}
+      run: |
+        cd workspace-manager-clienttests
+        echo "Upload test results to BigQuery Test Runner Dataset"
+        ./gradlew uploadResults --args="SummariesToBigQuery.json /tmp/TestRunnerResults"
```

Comment:
I'd kinda prefer we make a variable for a named directory for the results folder for every run so we don't accidentally upload stale results from a previous one. (Granted the files inside that directory are named with timestamps).

Should this step or a cleanup step delete the output dir after successful upload?

Reply:
Each runner starts in a fresh virtual environment, so this is handled automatically without us cleaning up.
(Side note, the github-provided cache step we use above is explicitly an exception to this)

Reply:
Sounds good.